File: moveZeroes.py
# see moveZeroes.js for comments/explanation
# working on py solutions
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zeroPos = []
        for i in range(len(nums)):
            if nums[i] == 0:
                zeroPos.append(i)
            else:
                if len(zeroPos) > 0:
                    zeroIdx = zeroPos.pop(0)
                    nums[zeroIdx] = nums[i]
                    zeroPos.append(i)
                    nums[i] = 0

File: moveZeroes.js
// https://leetcode.com/problems/move-zeroes/

/* Given an integer array nums, move all 0's to the end of it while maintaining the relative order of the non-zero elements.

Note that you must do this in-place without making a copy of the array.

*/

/**
 * param {number[]} nums
 * return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
    // step through the array
    // note the pos of encountered 0, append to arr
    let zeroPos = []
    nums.forEach((num, idx) => {
        // when zero found, push idx to array
        if (num === 0) {
            zeroPos.push(num)
        }

    });
    // when non-zero encountered, swap with lowest zero

};